contracts/strategies/StrategyProxy.sol

```diff
@@ -0,0 +1,101 @@
+    address public constant mintr = address(0xd061D61a4d941c39E5453435B6345Dc261C2fcE0);
+    address public constant crv = address(0xD533a949740bb3306d119CC777fa900bA034cd52);
+    address public constant gauge = address(0x2F50D538606Fa9EDD2B11E2446BEb18C9D5846bB);
+    address public constant y = address(0xFA712EE4788C042e2B7BB55E6cb8ec569C4530c1);
+
+    mapping(address => bool) public strategies;
+    address public governance;
+
+    constructor() public {
+        governance = msg.sender;
+    }
+
+    function setGovernance(address _governance) external {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function approveStrategy(address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        strategies[_strategy] = true;
+    }
+
+    function revokeStrategy(address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        strategies[_strategy] = false;
+    }
+
+    function lock() external {
+        proxy.increaseAmount(IERC20(crv).balanceOf(address(proxy)));
+    }
+
+    function vote(address _gauge, uint256 _amount) public {
+        require(strategies[msg.sender], "!strategy");
+        proxy.execute(gauge, 0, abi.encodeWithSignature("vote_for_gauge_weights(address,uint256)", _gauge, _amount));
+    }
+
+    function max() external {
+        require(strategies[msg.sender], "!strategy");
+        vote(y, 10000);
+    }
+
+    function withdraw(
+        address _gauge,
+        address _token,
+        uint256 _amount
+    ) public returns (uint256) {
+        require(strategies[msg.sender], "!strategy");
+        uint256 _before = IERC20(_token).balanceOf(address(proxy));
+        proxy.execute(_gauge, 0, abi.encodeWithSignature("withdraw(uint256)", _amount));
+        uint256 _after = IERC20(_token).balanceOf(address(proxy));
+        uint256 _net = _after.sub(_before);
+        proxy.execute(_token, 0, abi.encodeWithSignature("transfer(address,uint256)", msg.sender, _net));
+        return _net;
+    }
+
+    function balanceOf(address _gauge) public view returns (uint256) {
+        return IERC20(_gauge).balanceOf(address(proxy));
+    }
+
+    function withdrawAll(address _gauge, address _token) external returns (uint256) {
+        require(strategies[msg.sender], "!strategy");
+        return withdraw(_gauge, _token, balanceOf(_gauge));
+    }
+
+    function deposit(address _gauge, address _token) external {
+        uint256 _balance = IERC20(_token).balanceOf(address(this));
+        IERC20(_token).safeTransfer(address(proxy), _balance);
+
+        _balance = IERC20(_token).balanceOf(address(proxy));
+        proxy.execute(_token, 0, abi.encodeWithSignature("approve(address,uint256)", _gauge, 0));
+        proxy.execute(_token, 0, abi.encodeWithSignature("approve(address,uint256)", _gauge, _balance));
+        proxy.execute(_gauge, 0, abi.encodeWithSignature("deposit(uint256)", _balance));
```

Comment:
This is the offending line it seems

Reply:
We are going to test this first, then fix?